File: app/src/main/java/net/tlalka/android/fiszki/model/dao/AbstractDao.java
package net.tlalka.android.fiszki.model.dao;

import java.sql.SQLException;
import java.util.Collection;

import android.util.Log;

import com.j256.ormlite.dao.BaseDaoImpl;
import com.j256.ormlite.support.ConnectionSource;

public abstract class AbstractDao<T, ID> extends BaseDaoImpl<T, ID> {

    protected AbstractDao(ConnectionSource connectionSource, Class<T> dataClass) throws SQLException {
        super(connectionSource, dataClass);
    }

    @Override
    public int create(T data) {
        try {
            return super.create(data);
        } catch (SQLException ex) {
            Log.e(this.getClass().getName(), "DB create exception", ex);
            return 0;
        }
    }

    @Override
    public int create(Collection<T> data) {
        try {
            return super.create(data);
        } catch (SQLException ex) {
            Log.e(this.getClass().getName(), "DB create exception", ex);
            return 0;
        }
    }

    @Override
    public int update(T data) {
        try {
            return super.update(data);
        } catch (SQLException ex) {
            Log.e(this.getClass().getName(), "DB update exception", ex);
            return 0;
        }
    }

    @Override
    public int delete(T data) {
        try {
            return super.delete(data);
        } catch (SQLException ex) {
            Log.e(this.getClass().getName(), "DB delete exception", ex);
            return 0;
        }
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/model/dao/AbstractDao.kt
package net.tlalka.fiszki.model.dao

import android.util.Log
import com.j256.ormlite.dao.BaseDaoImpl
import com.j256.ormlite.support.ConnectionSource
import java.sql.SQLException

abstract class AbstractDao<T, ID>(connectionSource: ConnectionSource, dataClass: Class<T>)
        : BaseDaoImpl<T, ID>(connectionSource, dataClass) {

    override fun create(data: T): Int {
        return try {
            super.create(data)
        } catch (ex: SQLException) {
            Log.e(this.javaClass.name, "DB create exception", ex)
            0
        }
    }

    override fun create(data: Collection<T>): Int {
        return try {
            super.create(data)
        } catch (ex: SQLException) {
            Log.e(this.javaClass.name, "DB create exception", ex)
            0
        }
    }

    override fun update(data: T): Int {
        return try {
            super.update(data)
        } catch (ex: SQLException) {
            Log.e(this.javaClass.name, "DB update exception", ex)
            0
        }
    }

    override fun delete(data: T): Int {
        return try {
            super.delete(data)
        } catch (ex: SQLException) {
            Log.e(this.javaClass.name, "DB delete exception", ex)
            0
        }
    }
}
